### PR TITLE
Implement a conditional skip marker

### PR DESCRIPTION
**Pull request by roecla**
#### Changes

Add a marker called `skip_if` that takes two arguments, a condition and a reason.
If the condition is True (or all conditions are True in case more than one `skip_if` marker is used) the task is skipped, else it is run normally. 

#### Todo

- [x] Reference issues which can be closed due to this PR with "Closes #x".
- [x] Add test cases
- [x] Register and implement the marker
- [x] Test the marker in sid-germany.  
- [x] Add a tutorial explaining the two skip decorators. 
- [x] Document PR in docs/changes.rst.

### CODE REVIEW

File: docs/tutorials/how_to_skip_tasks.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    def task_always(produces):
+        ...
+
+    # Content of task_long_running.py
+
+    from config import FAST_FLAG
+
+    @pytask.mark.skipif(FAST_FLAG, "Fast Flag active")
+    @pytask.mark.depends_on("dependency_of_long_running_task.md")
+    def task_that_takes_really_long_to_run(depends_on):
```
Comment (tobiasraabe) [inline docs/tutorials/how_to_skip_tasks.rst:59]: Maybe this is a little bit more expressive. And I added code-blocks for each file.

```suggestion
.. code-block:: python

    # Content of a config.py

    NO_LONG_RUNNING_TASKS = True

.. code-block:: python

    # Content of task_create_dependency.py

    @pytask.mark.produces("run_always.md")
    def task_always(produces):
        ...

.. code-block:: python

    # Content of task_long_running.py

    from config import NO_LONG_RUNNING_TASKS
    
    @pytask.mark.skipif(NO_LONG_RUNNING_TASKS, "Skip long-running tasks.")
    @pytask.mark.depends_on("dependency_of_long_running_task.md")
    def task_that_takes_really_long_to_run(depends_on):
        ...

```

**Review by tobiasraabe — APPROVED** [review]
Hi Klara, the code is really well written, has tests and documentation! Thanks for great work. 🎉 

And thanks for being the first contributor! 🎆 

I added some comments to discuss some details. And could you add this PR to the Changes.rst as well as mentioning your handle with``:ghuser:`roecla` ``?

File: docs/tutorials/how_to_skip_tasks.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+.. code-block:: python
+
+    # Content of task_create_dependency.py
+
+    @pytask.mark.produces("dependency_of_long_running_task.md")
+    def task_you_are_working_on(produces):
+        ...
+
+
```
Comment (tobiasraabe) [inline docs/tutorials/how_to_skip_tasks.rst:25]:
```suggestion
.. code-block:: python

```

**Comment by asottile** [comment]
(sorry for drive-by) apologies for the slow pre-commit.ci run -- github gave us repeated 500s while trying to look up pull request information for this PR (and exhausted our retries).  I've requeued it and it looks like it succeeded now.

**Comment by tobiasraabe** [comment]
That is quite a suprise :). Thanks for stopping by and while you are here, thanks for pre-commit. A minor hiccup is nothing I think too long about given the tremendous benefits I get from the tool.

**Pull request merged**